presto-docs/src/main/sphinx/security/authorization.rst

```diff
@@ -0,0 +1,130 @@
+---------------------------
+
+Environment Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You must make the following changes to the environment prior to configuring the
+Presto authorization support.
+
+ * :ref:`internal-communication`
+ * :ref:`enable_authentication`
+
+.. _enable_authentication:
+
+Enable Authentication
+~~~~~~~~~~~~~~~~~~~~~
+
+Presto authorization requires authentication to get the accessor's principal,
+so make sure you have authentication enabled for Presto.
+
+   - If TLS/SSL is configured properly, we can just use the certificate to
+     identify the accessor.
+
+     .. code-block:: none
+
+         http-server.authentication.type=CERTIFICATE
+
+   - It is also possible to specify other authentication types such as
+     ``KERBEROS``, ``PASSWORD`` and ``JWT``. Additional configuration may be
+     needed.
+
+     .. code-block:: none
+
+         node.internal-address=<authentication type>
+
+Server Config Properties
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following is an example of the properties that need to be added to the
+``config.properties`` file:
+
+.. code-block:: none
+
+    http-server.authorization.enabled=true
+    http-server.authorization.default-policy=ALLOW
+    http-server.authorization.default-allowed-roles=USER,ADMIN
+    http-server.authorization.allow-unsecured-requests=false
+
+======================================================= ======================================================
+Property                                                Description
+======================================================= ======================================================
+``http-server.authorization.enabled``                   Enable authorization for the Presto.
+                                                        Should be set to ``true``. Default value is
+                                                        ``false``.
+``http-server.authorization.default-policy``            The default authorization policy applies to endpoints
+                                                        without allowed roles specified. Can be set to
+                                                        ``ALLOW``, ``DENY`` and ``DEFAULT_ROLES``.
+``http-server.authorization.default-allowed-roles``     The roles allowed to access the endpoints without
+                                                        explicitly specified when default-policy is set to
+                                                        ``DEFAULT_ROLES``.
+``http-server.authorization.allow-unsecured-requests``  Skip authorization check for unsecured requests.
+                                                        Default value is ``false``.
+======================================================= ======================================================
+
+.. warning::
+
+    Enable allowing unsecured requests only when you leave HTTP enabled and
+    need the HTTP endpoints to be accessible. In most cases this is a security
+    hole.
```

Comment:
`http-server.authorization.allow-unsecured-requests` is provided as a way to transition from HTTP to HTTPS with authorization and is a security hole since it allows unauthenticated requests to skip authorization checks. Only enable during the transition period and disable this setting once done.